pkg/controller/manager.go

```diff
@@ -89,11 +89,12 @@ func (m *Manager) UpdateController(name string, params ControllerParams) *Contro
 		ctrl.getLogger().Debug("Controller update time: ", time.Since(start))
 	} else {
```

Comment:
I can't comment on the correct line (75), but the `if exists` may also need to check `terminated` and recreate the channel. Otherwise, `UpdateController` will succeed, and everything will "work" until we re-use a run-once controller (such as the CNP updater)

Reply:
Discussed offline. The terminated channel is bound to the lifecycle of the goroutine which is never stopped until the controller is removed from the manager.